gammapy/datasets/flux_points.py

```diff
@@ -122,11 +139,52 @@ def __init__(
         self.models = models
         self.meta_table = meta_table
 
-        if mask_safe is None:
-            mask_safe = (~data.is_ul).data
+        self.available_stat_type = dict(
+            chi2=self._stat_array_chi2,
+            profile=self._stat_array_profile,
+            distrib=self._stat_array_distrib,
+        )
+
+        if stat_kwargs is None:
+            stat_kwargs = dict()
+        self.stat_kwargs = stat_kwargs
 
+        self.stat_type = stat_type
+
+        if mask_safe is None:
```

Comment:
In a different PR, maybe we should make the masks as Maps to be consistent with the `MapDataset`